agent/rpi_helper.py

```diff
@@ -19,3 +22,31 @@ def detect_raspberry_pi():
                 metadata['serial_number'] = line.split()[-1]
 
     return metadata
+
+
+def detect_confinement():
+    """
+    Returns the confinement environment.
+    This function is used to control features.
+    """
+
+    metadata = {
+        'docker': None,
+        'balena': None,
+        'snap': None,
+    }
```

Comment:
This logic will work, but all those fields will either be `None` or `True`. Not quite straight-forward.

Reply:
Also, this asks for an enum. For example the function can return `[Confinement.DOCKER, Confinement.BALENA]` and you can then use it like this:
```
if Confinement.DOCKER in confinement:
...
```

Reply:
I've tweaked this  bit per your comment (with False as defaults). `enum` looks like a good approach, but let's stick to this for now unless you disagree?

Reply:
I disagree. If we're the ones defining the possible values and we know all possible values we should use enum. It is kind of another static check, helps to avoid future errors, analyzer and intellisense-friendly.
Otherwise this is starting to look like Javascript.

Reply:
Ok, I've refactored this to use `ENUM`. Let me know if this makes more sense to you.

Reply:
@vpetersson You're using it wrong. You're defining enum members to be True or False. Enum is supposed to be a constant. You compare variables to this constant.

Let me rewrite this chunk if you don't mind.
```
class Confinement(Enum):
    DOCKER = 1
    BALENA = 2
    SNAP = 3

def detect_confinement():
    if os.getenv('SNAP'):
        return Confinement.SNAP
    is_docker = 'docker' in open('/proc/1/cgroup', 'rt').read()
    if is_docker and (os.getenv('BALENA') or os.getenv('RESIN')):
        return Confinement.BALENA
    else:
        return Confinement.DOCKER

### USAGE ###
confinement = detect_confinement()
if confinement is Confinement.SNAP:
# ...
````